golem-worker-service-base/src/service/mod.rs

```diff
@@ -1,6 +1,23 @@
+use golem_common::model::TemplateId;
+use golem_service_base::service::auth::Permission;
+
 pub mod api_definition;
 pub mod api_definition_validator;
-pub mod error;
 pub mod http_request_definition_lookup;
 pub mod template;
 pub mod worker;
+
+#[derive(Clone, Debug)]
+pub struct TemplatePermission {
+    pub template: TemplateId,
+    pub permission: Permission,
+}
```

Comment:
Fine grained approach -that we aleays discussed. Its nice, but keep a general principle that a fine grained approach often result in too-many possibilities, of which some of them could be illegal. Example, what does it mean to have a product of `Permission::Create` for a `TemplateId` that already exist

Reply:
So if keeping fine-grained approach, the `Permission` should be different type too, to disallow bumping into this property. For an existing `TemplateId` , only `View` , `Update` and `Delete` is possible